src/main/kotlin/simplecolocalization/ImageJOpener.kt

```
@@ -0,0 +1,68 @@
+package simplecolocalization
+
+import ij.IJ
+import ij.ImagePlus
+import ij.gui.MessageDialog
+import ij.io.Opener
+import ij.process.ByteProcessor
+import java.io.File
+import java.lang.Integer.min
+import loci.formats.`in`.LIFReader
+
+class ImageJOpener {
+
+    private val opener = Opener()
+    private val lifReader = LIFReader()
+    private lateinit var extension: String
+    private lateinit var absolutePath: String
+    private var isLif: Boolean = false
+    private var isTiff: Boolean = false
+
+    private fun init(file: File) {
+        extension = file.extension
+        absolutePath = file.absolutePath
+        if (extension == "lif") isLif = true
+        if (extension == "tiff" || extension == "tif") isTiff = true
+    }
+
+    fun openStack(file: File, numSlices: Int): List<ImagePlus>? {
```

Instead of returning null here upon error, what you should instead do is throw an exception, maybe our own `UnsupportedFileType` exception which extends `IllegalArgumentType`.

This is partly because this class should not be responsible for the UI (i.e., SimpleColocalization should be responsible for opening message dialogues) and also because the user of this class may assume the output is non-null and will continue doing their processing even if the file is unsupported.